moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_calcs.h

```diff
@@ -77,30 +83,44 @@ class JogCalcs
 
   sensor_msgs::JointState incoming_joints_;
 
+  /** \brief Do jogging calculations for Cartesian twist commands. */
   bool cartesianJogCalcs(geometry_msgs::TwistStamped& cmd, JogArmShared& shared_variables, std::mutex& mutex);
 
-  bool jointJogCalcs(const control_msgs::JointJog& cmd, JogArmShared& shared_variables);
+  /** \brief Do jogging calculations for direct commands to a joint. */
+  bool jointJogCalcs(const control_msgs::JointJog& cmd, JogArmShared& shared_variables, std::mutex& mutex);
 
-  // Parse the incoming joint msg for the joints of our MoveGroup
+  /** \brief Update the stashed status so it can be retrieved asynchronously */
+  void updateCachedStatus(JogArmShared& shared_variables, std::mutex& mutex);
```

Comment:
Why not put the mutex in the JogArmShared or maybe even use setters / getters for accessing the shared_variables that manage locking / unlocking the mutex internally?

Reply:
fixed by making the status code atomic